src/parallel_progress.jl

```diff
@@ -0,0 +1,376 @@
+    pp = ParallelProgress(channel)
+    
+    @async begin
+        try
+            while !has_finished(progress) && !has_finished(pp)
+                f, args, kw = take!(channel)
+                if f == PP_NEXT
+                    next!(progress, args...; kw...)
+                elseif f == PP_CANCEL
+                    cancel(progress, args...; kw...)
+                    break
+                elseif f == PP_FINISH
+                    finish!(progress, args...; kw...)
+                    break
+                elseif f == PP_UPDATE
+                    update!(progress, args...; kw...)
+                end
+            end
+        catch err
+            println()
+            # channel closed should only happen from Base.close(pp), which isn't an error
+            if err != Base.closed_exception()
+                bt = catch_backtrace()
+                showerror(stderr, err, bt)
+                println()
+            end
+        finally
+            close(pp)
+        end
+    end
+    return pp
+end
+
+
+"""
+    FakeChannel()
+
+fake RemoteChannel that doesn't put anything anywhere (for allowing overshoot)
+"""
+struct FakeChannel end
+Base.close(::FakeChannel) = nothing
+Base.isready(::FakeChannel) = false
+Distributed.put!(::FakeChannel, _...) = nothing
+
+struct MultipleChannel
+    channel
+    id
+end
+Distributed.put!(mc::MultipleChannel, x) = put!(mc.channel, (mc.id, x...))
+
+mutable struct MultipleProgress
+    channel
+    amount::Int
+end
+
+Base.getindex(mp::MultipleProgress, n) = ParallelProgress.(MultipleChannel.(Ref(mp.channel), n))
+Base.lastindex(mp::MultipleProgress) = mp.amount
+
+"""
+    MultipleProgress(progresses::AbstractVector{<:AbstractProgress},
+                     mainprogress::AbstractProgress;
+                     enabled = true,
+                     auto_close = true,
+                     count_finishes = false,
+                     count_overshoot = false,
+                     auto_reset_timer = true)
+
+allows to call the `progresses` and `mainprogress` from different workers
+ - `progresses`: contains the different progressbars
+ - `mainprogress`: main progressbar
+ - `enabled`: `enabled == false` doesn't show anything and doesn't open a channel
+ - `auto_close`: if true, the channel will close when all progresses are finished, otherwise,
+ when mainprogress finishes or with `close(p)`
+ - `count_finishes`: if false, main_progress will be the sum of the individual progressbars,
+ if true, it will be equal to the number of finished progressbars
+ - `count_overshoot`: overshooting progressmeters will be counted in the main progressmeter
+ - `auto_reset_timer`: tinit in progresses will be reset at first call
+
+use p[i] to access the i-th progressmeter, and p[0] to access the main one
+
+# Example
+```jldoctest
+julia> using Distributed
+julia> addprocs(2)
+julia> @everywhere using ProgressMeter
+julia> p = MultipleProgress(fill(10,5); desc="global ", kws=[(desc="task \$i ",) for i in 1:5])
```

Comment:
`fill(10,5)` doesn't match the signature in the docstring?

Reply:
yes, this doesn't work anymore, I'll fix the docstring